### PR TITLE
Updated the LIEF worker plugin to the latest stoQ API.

### PR DESCRIPTION
**Pull request by serializingme**
As promised, the updated version of the LIEF worker plugin. Let me know what you think and if there are any changes that need to be made.

### CODE REVIEW

**Comment by serializingme** [comment]
I'm looking into adding the tests to the plugin and I want to ask what is (from your perspective) the best approach to obtain the test data:

1. Create a submodule on the `data` directory pointing to existing binary samples repositories, for example, [LIEF test samples](https://github.com/lief-project/samples) or [JonathanSalwan Binary Samples](https://github.com/JonathanSalwan/binary-samples)
1. Copy the binary samples from such repositories, but include the respective licensing information
1. Create stoQ own collection on a separate repository (e.g. `stoq-samples`) that would be linked to the `data` directory as a submodule and that could be used by other plugins like `peinfo`

Even though it would take longer, I'm inclined to go for the 3rd option.

File: lief/lief/lief.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+class LiefPlugin(WorkerPlugin):
+    def __init__(self, config: ConfigParser, plugin_opts: Optional[Dict]) -> None:
+        super().__init__(config, plugin_opts)
+
+        if plugin_opts and 'abstract' in plugin_opts:
+            abstract = bool(plugin_opts['abstract'])
+        elif config.has_option('options', 'abstract'):
+            abstract = bool(config.get('options', 'abstract'))
+        self.abstract = abstract
```
Comment (mlaferrera) [inline lief/lief/lief.py]: I would recommend setting a default for all required variables within plugins. In this case, this plugin will raise an exception if `abstract` is not defined within plugin options or the plugin configuration. 

As a side note, I realize several other plugins do not follow this behavior, it's something I plan on fixing in the near future.

File: lief/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# lief
+
+[stoQ](https://stoq-framework.readthedocs.io/en/latest/index.html) plugin that parses and abstracts PE, ELF and MachO executable files using LIEF and produces metadata about the payload.
```
Comment (mlaferrera) [inline lief/README.md]: Do you mind adding an inline link to `LIEF` so potential users can check it out and get a better understanding of what it offers?

**Comment by serializingme** [comment]
I have addressed your comments :D I will create an issue for the binary / file samples discussion.

**Pull request merged**